lib/cpp/Kconfig

```diff
@@ -56,6 +56,55 @@ config STD_CPP2B
 
 endchoice
 
+config REQUIRES_FULL_LIBCPP
+	bool
+	select REQUIRES_FULL_LIBC
+	help
+	  Helper symbol to indicate that a full C++ standard library
+	  implementation is required.
+
+	  Select this from a symbol that enables a feature requiring a full
+	  C++ standard library implementation.
+
+choice LIBCPP_IMPLEMENTATION
+	prompt "C++ Standard Library Implementation"
+	default EXTERNAL_LIBCPP if REQUIRES_FULL_LIBCPP && NATIVE_APPLICATION
+	default GLIBCXX_LIBCPP if REQUIRES_FULL_LIBCPP
+	default MINIMAL_LIBCPP
```

Comment:
any way to get ARCMWDT_LIBCPP selected when the user selects ARCMWDT_LIBC?

Reply:
That is technically possible by adding `default ARCMWDT_LIBCPP if REQUIRES_FULL_LIBCPP && ARCMWDT_LIBC`, but I opted not to do that in this PR because I do not know enough about the ARC MWDT toolchain to make that assumption and I do not have access to that toolchain either.

@evgeniy-paltsev @abrodkin Does the above patch sound reasonable? If so, I can add it this PR; alternatively, you can send a follow-up patch after this PR is merged if you are not sure at this time.

Reply:
> There is no `ARCMWDT_LIBCPP `currently, the logic for selecting the Metaware C++ library is `(CONFIG_ARCMWDT_LIBC AND CONFIG_LIB_CPLUSPLUS)`. So clearly some logic to repair this has to be added.

No logic repair should be necessary.

The C++ standard library inclusion is currently handled through `toolchain_ld_cpp`:

https://github.com/zephyrproject-rtos/zephyr/blob/85a7d0cc467384be33ce7a6d80f35cefce1d6d1b/CMakeLists.txt#L342-L345

https://github.com/zephyrproject-rtos/zephyr/blob/85a7d0cc467384be33ce7a6d80f35cefce1d6d1b/cmake/linker/arcmwdt/target.cmake#L194-L199

After the changes in this PR, `toolchain_ld_cpp` is called `if (NOT CONFIG_MINIMAL_LIBCPP)`, so it will still work when `CONFIG_ARCMWDT_LIBCPP` is selected.

This `toolchain_ld_cpp` function could be reworked to more verbosely handle the C++ standard library linkage based on the selected LIBCPP config, but that is something for the future improvements.

Reply:
ok, if MWDT C++ still works we can improve in a separate PR. @evgeniy-paltsev, please confirm this doesn't break MWDT C++.

Reply:
@evgeniy-paltsev Have you had a chance to test this?